src/bootstrap/native.rs

```diff
@@ -416,13 +416,18 @@ impl Step for Llvm {
             //        should use llvm-tblgen from there, also should verify that it
             //        actually exists most of the time in normal installs of LLVM.
             let host_bin = builder.llvm_out(builder.config.build).join("bin");
-            cfg.define("CMAKE_CROSSCOMPILING", "True");
             cfg.define("LLVM_TABLEGEN", host_bin.join("llvm-tblgen").with_extension(EXE_EXTENSION));
-            cfg.define("LLVM_NM", host_bin.join("llvm-nm").with_extension(EXE_EXTENSION));
             cfg.define(
                 "LLVM_CONFIG_PATH",
                 host_bin.join("llvm-config").with_extension(EXE_EXTENSION),
             );
+            if builder.config.llvm_clang {
+                let build_bin = builder.llvm_out(builder.config.build).join("build").join("bin");
+                cfg.define(
+                    "CLANG_TABLEGEN",
+                    build_bin.join("clang-tblgen").with_extension(EXE_EXTENSION),
+                );
+            }
```

Comment:
not quite sure why this is necessary, but seems harmless - can you add an assert that this file exists?

Reply:
Similar to `llvm-tblgen`, `clang-tblgen` is used to generate code when building `clang`. Do we also want to add assertions for other files during cross compilation? (`llvm-tblgen`, `llvm-config`)

Reply:
I think it's ok to only add an assert if we're explicitly setting the variable - I've had bad experiences with cmake and don't trust it to report errors in a way that make sense, but clang is usually pretty good at telling you if things aren't found.

Reply:
I added the file check and assertion as requested.